jupyterhub/handlers/base.py

```diff
@@ -771,8 +771,9 @@ async def auth_to_user(self, authenticated, user=None):
         # Only set `admin` if the authenticator returned an explicit value.
         if admin is not None and admin != user.admin:
             user.admin = admin
-            roles.assign_default_roles(self.db, entity=user)
-            self.db.commit()
+        # always ensure default roles ('user', 'admin' if admin) are assigned
+        # after a successful login
+        roles.assign_default_roles(self.db, entity=user)
```

Comment:
We will now no longer make the self.db.commit() call after having modified a user model, is that alright?

Reply:
it was redundant, `assign_default_roles` already contained the necessary commits if any changes are made.